cpp/include/cudf/contiguous_split.hpp

```diff
@@ -0,0 +1,229 @@
+   */
+  packed_columns(std::unique_ptr<metadata>&& md, std::unique_ptr<rmm::device_buffer>&& gd)
+    : metadata_(std::move(md)), gpu_data(std::move(gd))
+  {
+  }
+
+  std::unique_ptr<metadata> metadata_;           ///< Host-side metadata buffer
+  std::unique_ptr<rmm::device_buffer> gpu_data;  ///< Device-side data buffer
+};
+
+/**
+ * @brief The result(s) of a cudf::contiguous_split
+ *
+ * @ingroup copy_split
+ *
+ * Each table_view resulting from a split operation performed by contiguous_split,
+ * will be returned wrapped in a `packed_table`.  The table_view and internal
+ * column_views in this struct are not owned by a top level cudf::table or cudf::column.
+ * The backing memory and metadata is instead owned by the `data` field and is in one
+ * contiguous block.
+ *
+ * The user is responsible for assuring that the `table` or any derived table_views do
+ * not outlive the memory owned by `data`
+ */
+struct packed_table {
+  cudf::table_view table;  ///< Result table_view of a cudf::contiguous_split
+  packed_columns data;     ///< Column data owned
+};
+
+/**
+ * @brief Performs a deep-copy split of a `table_view` into a set of `table_view`s into a single
+ * contiguous block of memory.
+ *
+ * @ingroup copy_split
+ *
+ * The memory for the output views is allocated in a single contiguous `rmm::device_buffer` returned
+ * in the `packed_table`. There is no top-level owning table.
+ *
+ * The returned views of `input` are constructed from a vector of indices, that indicate
+ * where each split should occur. The `i`th returned `table_view` is sliced as
+ * `[0, splits[i])` if `i`=0, else `[splits[i], input.size())` if `i` is the last view and
+ * `[splits[i-1], splits[i]]` otherwise.
+ *
+ * For all `i` it is expected `splits[i] <= splits[i+1] <= input.size()`
+ * For a `splits` size N, there will always be N+1 splits in the output
+ *
+ * @note It is the caller's responsibility to ensure that the returned views
+ * do not outlive the viewed device memory contained in the `all_data` field of the
+ * returned packed_table.
+ *
+ * @code{.pseudo}
+ * Example:
+ * input:   [{10, 12, 14, 16, 18, 20, 22, 24, 26, 28},
+ *           {50, 52, 54, 56, 58, 60, 62, 64, 66, 68}]
+ * splits:  {2, 5, 9}
+ * output:  [{{10, 12}, {14, 16, 18}, {20, 22, 24, 26}, {28}},
+ *           {{50, 52}, {54, 56, 58}, {60, 62, 64, 66}, {68}}]
+ * @endcode
+ *
+ *
+ * @throws cudf::logic_error if `splits` has end index > size of `input`.
+ * @throws cudf::logic_error When the value in `splits` is not in the range [0, input.size()).
+ * @throws cudf::logic_error When the values in the `splits` are 'strictly decreasing'.
+ *
+ * @param input View of a table to split
+ * @param splits A vector of indices where the view will be split
+ * @param[in] mr Device memory resource used to allocate the returned result's device memory
+ * @return The set of requested views of `input` indicated by the `splits` and the viewed memory
+ * buffer.
```

Comment:
Don't use period `.` inconsistently.
```suggestion
 * buffer
```